dask/array/tests/test_array_core.py

```diff
@@ -2286,6 +2286,24 @@ def test_astype():
     assert d.astype("f8") is d
 
 
+def test_astype_gh1151():
+    a = np.arange(5).astype(np.int32)
+    b = da.from_array(a, (1,))
+    assert_eq(a.astype(np.int16), b.astype(np.int16))
+
+
+def test_astype_gh9318():
+    # `order`` kwarg in `astype` should not cause an error
+    a = np.array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12]], order="C")
+    b = da.from_array(a, chunks=(2, 2))
+    a.astype(float, order="F")
+    b.astype(float, order="F")  # if no error at this line, pytest passes
+    # This assertion currently failing because of issue https://github.com/dask/dask/issues/9316
+    # result = b.compute()
+    # assert a.flags.c_contiguous == result.flags.c_contiguous
+    # assert a.flags.f_contiguous == result.flags.f_contiguous
```

Comment:
I don't know what the preference from maintainers is - leave this commented outline for a better test someday, or remove the comments entirely. I can remove these lines if need be.

Reply:
Would an XFAIL test be the right way to go here? @jrbourbeau do you have any thoughts?

Reply:
Yeah, nice suggestion @ncclementi. I'd recommend moving the currently commented out portion to a separate test and then mark it as `xfail`. Since we're using [strict `xfail`ing](https://github.com/dask/dask/blob/50ab8af982a31b186a8e47597f0ad7e5b59bcab5/setup.cfg#L64) that means when https://github.com/dask/dask/issues/9316 is fixed this new test will start failing, otherwise we may forget to come back here and comment these lines once https://github.com/dask/dask/issues/9316 is resolved

Reply:
Done.
A heads up that https://github.com/dask/dask/issues/9316 is probably **not** going to get fixed. It's a waste of resources to spend time on it when it's not currently causing a problem for anyone - see discussion here https://github.com/dask/dask/issues/9316#issuecomment-1202762657

Given that, if it makes more sense to take the XFAIL test out of the codebase, you can go ahead and do that too. I generally don't mind if my PRs are edited by other people, especially if someone wants to make a quick change so they can hit the merge button without a extra delays.